Source Chargebee: Updates schemas for validation and missing fields errors

pnilan opened the pull request:
## What
- Sets `additionalProperties` to true for `custom_fields` objects for each stream
- Adds missing fields for streams, per dashboard
- Updates fields for schema validation errors, per dashboard
- Updates changelog

artem1205 commented on airbyte-integrations/connectors/source-chargebee/source_chargebee/schemas/customer.json:
```diff
@@ -175,6 +175,15 @@
+    "tax_providers_fields": {
+      "type": ["array", "null"],
+      "items": {
+        "type": ["integer", "string", "null"]
```
Type cannot be declared in that way

pnilan replied:
@artem1205 Is this due to the way Airbyte handles converting data types?

artem1205 replied:
Yep, this may cause an issue in destination connectors

pnilan replied:
Removed in latest commit

artem1205 commented on airbyte-integrations/connectors/source-chargebee/source_chargebee/schemas/customer.json:
```diff
@@ -431,6 +440,7 @@
       "type": ["null", "array"],
       "items": {
         "type": ["null", "object"],
+        "additionalProperties": true,
```
No need to add additionalProperties: true for each object. this is a default behaviour

pnilan replied:
Removed in latest commit. Thanks for the insight.

artem1205 commented on airbyte-integrations/connectors/source-chargebee/source_chargebee/schemas/payment_source.json:
```diff
@@ -152,7 +152,7 @@
         "funding_type": {
           "type": ["string", "null"],
-          "enum": ["credit", "dedit", "prepaid", "not_known", "not_applicable"]
+          "enum": ["credit", "debit", "prepaid", "not_known", "not_applicable"]
```
enum can also be removed

pnilan replied:
@artem1205 Would you recommend I remove enumeration validation from all schemas for consistency? Looks like it was originally included when the connector was developed. I imagine this would only cause validation issues on our end.

pnilan replied:
Removed all unnecessary validation properties in the latest commit.

artem1205 commented on airbyte-integrations/connectors/source-chargebee/source_chargebee/schemas/comment.json:
```diff
@@ -28,7 +28,19 @@
     },
     "custom_fields": {
-      "type": ["array", "null"]
+      "type": ["null", "array"],
+      "items": {
+        "type": ["null", "object"],
+        "additionalProperties": true,
+        "properties": {
+          "name": {
+            "type": ["null", "string"]
+          },
+          "value": {
+            "type": ["null", "string"]
+          }
+        }
+      }
```
please move custom field to `schema/shared` folder and make a ref to it in all schemas.

pnilan replied:
Moved to shared/_definitions.json and references added in each schema

artem1205 reviewed (changes requested):
To resolve https://github.com/airbytehq/airbyte/issues/24553 you have to remove fail_on_extra_colmns flag from acceptance test config

pnilan commented:
@artem1205 Made some changes here:

- After discussion with Maxime decided to bypass the `subscription`, `customer`, and `coupon` streams in `acceptance-test-config` and test these streams via integration testing as we're moving towards relaxing CATs.
- Added semi-incremental support for `unbilled_charge, `site_migration_detail`, and `gift` streams as these provide cursor fields but no inherent filtering/sorting methods. Added a semi incremental component and additional unit testing for the component.
